app/api/currentuser/send_group_leave_request.go

```diff
@@ -0,0 +1,48 @@
+package currentuser
+
+import (
+	"net/http"
+
+	"github.com/France-ioi/AlgoreaBackend/app/service"
+)
+
+// swagger:operation POST /current-user/group-leave-requests/{group_id} groups users groupLeaveRequestCreate
+// ---
+// summary: Create a request to leave a group
+// description: >
+//   Lets a user create a request to leave a group.
+//
+//     On success the service creates a new row in `group_pending_requests` with `group_id` = `{group_id}`
+//     `type` = 'leave_request' and `member_id` = user's `group_id` + a new row in `group_membership_changes`
+//     with `action` = `leave_request_created` and `at` equal to current UTC time.
+//
+//     * If there is already a row in `groups_groups` or a row in `group_pending_request` with
+//       `type` == 'leave_request', the "unchanged" (201) response is returned.
```

Comment:
I assume it means you do not change anything in this case.
Actually (probably there is the similar cases with invitations and join requests), no groups_groups with a leave pending request is a bit of a problem. Even if it is not possible if we used the services only and there is no bug (anyway with expiration, probably it can happen), something enforces that by design, so we cannot assume it cannot happen.
Good it is what you do here, but I think in this case (of groups leave,join requests, and invitations) we should find a solution which "heals" the state instead of blocking on it.
For instance, we could say that if there is no membership but a pending leave request, we could remove the request (in practice, it means that if there is no membership, you should just silently delete all pending leave request)

Reply:
It's impossible to create a "leave request" while not being a group member. I'm not sure we should delete something in this case.

Reply:
> (anyway with expiration, probably it can happen)

GroupGroup.Transitions() ignore the `expires_at` column, so it cannot happen actually

Reply:
"Impossible" in the services, not by design. So manual changes (or "bugs") could create inconsistencies. Self-healing transient inconsistencies are not such of a problem, here inconsistencies may block operations on the membership of concern.

(ok that is not about this part of code anymore as it has been fixed)

Reply:
fixed